### DISCUSSION TITLE
Help me get the link of this channel: http://fr.cross.tv/channel/22405/

### DISCUSSION THREAD

**Discussion by Djane** [iptv-org/community#173] [Q&A] [answered]
Hello @LaneSh4d0w , @Sphinxroot , @keystroke3
Can you help me get the dedicated gospel music channel link?
I tried like below but it doesn't work

#EXTINF: 0, Vine TV (test 1)
https://query-streamlink.herokuapp.com/iptv-query?streaming-ip=http://fr.cross.tv/channel/22405/
#EXTINF: 0, Vine TV (test 2)
https://query-streamlink.herokuapp.com/iptv-query?streaming-ip=http://fr.cross.tv/channel/22405/live

Thank you

**Comment by Dum4G** [comment]
Got anathemed for whatever reason
![image](https://user-images.githubusercontent.com/5430116/134053735-31bf496c-76c7-4e28-a7b8-8087bcb081c8.png)

**Comment by keystroke3** [comment]
Here, but this will not last that long:
```
http://hls-live-linear-web-media.cdn01.net/LiveChannels/stream_22405/chunklist.m3u8?SYw0ZQk-hGzpF-vEBOAd3YZ7hqhSXlUeesOueRPzSbR1xN6U6Yvo9mdqrsUhUlmkI6guVnp5Iw12KuxY4jNXBGeoyovQpoQOyeI9xoEedmhzgTkEJNzoyPMiSKnN3iTRDTESHjma7-j9hUAwBaq9KrDEt4n5OpvezOS4W1n5cb77Nn3UCFW6s1jB4J1UVQVrm6TVXFFTXRxWUiyBqYaWKi0hvwA=&channel_id=22405&source=crosstv&device_id=0725671952-863c72440376b4692a285e0e0210a308&lcp=2&client_id=508966495297823132&play_id=1632163688990286
```

**Comment by keystroke3** [comment]
From the looks of it, the first query is what gives access. You can change all the other query parameters but not the first one. Getting a valid first query parameter would require looking at the js that generates it and that is assuming it is client generated.
EDIT: Actually, you don't need all the other parameters, just the first one:
```
http://hls-live-linear-web-media.cdn01.net/LiveChannels/stream_22405/chunklist.m3u8\?SYw0ZQk-hGzpF-vEBOAd3YZ7hqhSXlUeesOueRPzSbR1xN6U6Yvo9mdqrsUhUlmkI6guVnp5Iw12KuxY4jNXBGeoyovQpoQOyeI9xoEedmhzgTkEJNzoyPMiSKnN3iTRDTESHjma7-j9hUAwBaq9KrDEt4n5OpvezOS4W1n5cb77Nn3UCFW6s1jB4J1UVQVrm6TVXFFTXRxWUiyBqYaWKi0hvwA
```

**Comment by BellezaEmporium** [comment]
> From the looks of it, the first query is what gives access. You can change all the other query parameters but not the first one. Getting a valid first query parameter would require looking at the js that generates it and that is assuming it is client generated.

You might be lucky enough if the HTML DOM is read successfully, and you could scrape it using some JS.

**Comment by keystroke3** [comment]
Is there a way to programmatically load the DOM and run the js? The player will not load without js. Yes, if we could or somehow trigger the request, we might scrape it. Just look for a request with .m3u8 in it and grab the link.

**Comment by BellezaEmporium** [comment]
I personally use node-fetch and sometimes get used of some regex in case things get tougher.

As for your question, something like Puppeteer/Selenium could do the trick.

**Comment by keystroke3** [comment]
Hmm, a headless browser seems a bit too much for my small server, especially if there is  a large number of queries. If @Djane  has a little know-how  and a computer, a local server can be created to perform these queries.

**Comment by keystroke3** [comment]
I can look into making something like this as a weekend project, but can't commit to anything right now.

**Comment by Sphinxroot** [comment]
maybe if you use this https://www.lightcast.com/embed/player.php?width=1024&height=720&id=22405&type=live  in streamlink it might work

**Comment by Djane** [comment]
Hello @Sphinxroot,
Not working this streamlink: https://www.lightcast.com/embed/player.php?width=1024&height=720&id=22405&type=live (Not supported!)

#Below, web source:
https://www.kitvnetwork.com/vine/
https://www.kitvnetwork.com/kitv-worship/

**Comment by BellezaEmporium** [comment]
> maybe if you use this https://www.lightcast.com/embed/player.php?width=1024&height=720&id=22405&type=live  in streamlink it might work

Could be interesting to dig in this. Can't do it right now since I'm onto various things and I'm abroad for working purposes, but I'll try to have a look very soon.

![Screenshot_20210921-093733__01.jpg](https://user-images.githubusercontent.com/30985701/134123237-19416bb1-f6bb-46f7-943e-56606779696e.jpg)

**Comment by Djane** [comment]
Noted dear @LaneSh4d0w 

Many thanks

**Comment by BellezaEmporium** [comment]
I'm gonna look at this, at the same time than some other channels. Wait and see.

EDIT : The player's source code is a huuuuuuuge pile of crappy stuff. However I've managed to get an interesting link. More soon.

**Reply by Djane** [reply]
Okay i'm patient

**Comment by BellezaEmporium** [comment, marked as answer]
https://iptv-all.lanesh4d0w.codes/united-states/kitv-vine
https://iptv-all.lanesh4d0w.codes/united-states/kitv-worship

EDIT : Doesn't seem to work outside of my local version. This means it needs to be proxied.

**Comment by Djane** [comment]
Hello @LaneSh4d0w,
I hope you're well.
Just a reminder about the streams link's bellow
Have a nice day

https://iptv-all.lanesh4d0w.codes/united-states/kitv-vine
https://iptv-all.lanesh4d0w.codes/united-states/kitv-worship

**Reply by BellezaEmporium** [reply]
> 
> 
> Hello @LaneSh4d0w, I hope you're well. Just a reminder about the streams link's bellow Have a nice day
> 
> https://iptv-all.lanesh4d0w.codes/united-states/kitv-vine https://iptv-all.lanesh4d0w.codes/united-states/kitv-worship

I think the stream links it was giving couldn't be used for another person due to token being generated per-IP and so locked the stream if IP didn't match the token.
